src/app/services/job.service.ts

```diff
@@ -1,79 +1,127 @@
+  public getLoading(): Observable<boolean> {
+    return this.loadingChanged.asObservable();
   }
-
+ 
   /**
-   * @description Getter for selected job index
-   * @returns {number} Returns the index of the selected job
+   * @description - Updates the loadingChanged BehaviorSubject. This is called within the service's call() to update the status of the API. The loading variable is updated to true when the API is called, and updated to false when the response is retrieved.
+   * @param {boolean} loading - The new value for loading
+   * @returns {void}
    */
-  public getSelectedIndex(): number {
-    return this.selectedJobIndex;
+  private updateLoading(loading: boolean): void {
+    this.loading = loading;
+    this.loadingChanged.next(this.loading);
+  }
+ 
+  /**
+   * @description - Returns the status of isSuccessfullyCompleted. This should be used in components to determine if an API should be called again.
+   * @returns {boolean} - true if the API has successfully been called; false otherwise
+   */
+  public hasSuccessfullyCompleted(): boolean {
+    return this.isSuccessfullyCompleted;
+  }
+ 
+  /**
+   * @description - Returns the <yourModel> API result
+   * @returns {JobsResponse} - The API results for the inquireBillingAuthorization call
+   */
+  public getResults(): JobsResponse {
+    return this.jobApiResults;
+  }
+ 
+  /**
+   * @description - Calls the back end in order to get current job data based on given uuid. For use in homepage to load job data when the app is loaded or refreshed.
+   * @param {string} uuid - the uuid to be passed to the back end as a seed to generate jobs.
+   * @returns {void}
+   */
+  public call(uuid: string): void {
+    // validate we're not already loading an API response and that we have the expected parameters
+    if (!this.loading && uuid) {
+      this.updateLoading(true);
+      this.httpSubscription = this.http.get(`${this.serverURL}/${uuid}`)
+        .subscribe({ 
+          next: (response: any) => {
+            this.jobApiResults = new JobsResponse(response);
+            this.isSuccessfullyCompleted = true;
+            this.updateLoading(false);
+          },
+          error: (error: any) => {
+            this.jobApiResults = new JobsResponse(error.error);
+            this.isSuccessfullyCompleted = false;
+            this.updateLoading(false);
+          }
+        });
+      }
   }
 
   /**
-   * @description Returns the loadingChanged Subject as an Observable. 
-   * @returns {Observable<JobData[]>} - The jobListChanged Subject as an Observable 
+   * @description Getter for currently selected job.
+   * @returns {JobData} Returns the selected job
    */
-  public getJobListChanged(): Observable<JobData[]> {
-    return this.jobListChanged.asObservable();
+   public getSelectedJob(): JobData {
+    if (!this.loading && this.isSuccessfullyCompleted) {
```

Comment:
this might protect against the billion subscriptions in every sub-view component

Reply:
Well said.